server/service/src/invoice_line/outbound_shipment_line/delete/mod.rs

```diff
@@ -76,3 +76,103 @@ where
+            mock_outbound_shipment_b_invoice_lines, mock_store_a, MockDataInserts,
+        },
+        test_db::setup_all,
+        InvoiceLineRowRepository,
+    };
+
+    use crate::{
+        invoice_line::outbound_shipment_line::delete::DeleteOutboundShipmentLine,
+        invoice_line::outbound_shipment_line::DeleteOutboundShipmentLineError as ServiceError,
+        service_provider::ServiceProvider,
+    };
+
+    #[actix_rt::test]
+    async fn delete_inbound_shipment_line_errors() {
+        let (_, _, connection_manager, _) = setup_all(
+            "delete_outbound_shipment_line_errors",
+            MockDataInserts::all(),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider.context().unwrap();
+        let service = service_provider.invoice_line_service;
+
+        // LineDoesNotExist
+        assert_eq!(
+            service.delete_outbound_shipment_line(
+                &context,
+                &mock_store_a().id,
+                DeleteOutboundShipmentLine {
+                    id: "invalid".to_owned(),
+                },
+            ),
+            Err(ServiceError::LineDoesNotExist)
+        );
+
+        //NotAnOutboundShipment
+        assert_eq!(
+            service.delete_outbound_shipment_line(
+                &context,
+                &mock_store_a().id,
+                DeleteOutboundShipmentLine {
+                    id: mock_inbound_shipment_a_invoice_lines()[0].id.clone(),
+                },
+            ),
+            Err(ServiceError::NotAnOutboundShipment)
+        );
+
+        //CannotEditFinalised
+        assert_eq!(
+            service.delete_outbound_shipment_line(
+                &context,
+                &mock_store_a().id,
+                DeleteOutboundShipmentLine {
+                    id: mock_outbound_shipment_b_invoice_lines()[1].id.clone(),
+                },
+            ),
+            Err(ServiceError::CannotEditFinalised)
+        );
+
+        //TODO DatabaseError, NotThisStoreInvoice
+    }
+
+    #[actix_rt::test]
+    async fn delete_outbound_shipment_line_success() {
+        let (_, connection, connection_manager, _) = setup_all(
+            "delete_outbound_shipment_line_success",
+            MockDataInserts::all(),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider.context().unwrap();
+        let service = service_provider.invoice_line_service;
+
+        let invoice_line_id = service
+            .delete_outbound_shipment_line(
+                &context,
+                &mock_store_a().id,
+                DeleteOutboundShipmentLine {
+                    id: mock_outbound_shipment_a_invoice_lines()[0].id.clone(),
+                },
+            )
+            .unwrap();
+
+        //test entry has been deleted
```

Comment:
This might not have been in the original test, but we want to test that stock line was adjusted back correctly.

*test1*

invoice -> picked

invoice_line
number_of_packs: 10

stock_line
available_number_of_packs:  5
total_number_of_packs: 10

•line deleted*

stock_line
available_number_of_packs: 15
total_number_of_packs: 20

*test2*


invoice -> new or allocated

invoice_line
number_of_packs: 5

stock_line
available_number_of_packs:  5
total_number_of_packs: 10

•line deleted*

stock_line
available_number_of_packs: 10
total_number_of_packs: 10